tests/0137-barrier_batch_consume.c

```diff
@@ -0,0 +1,290 @@
+                            testid);
+                rd_kafka_message_destroy(rkmessage[i]);
+        }
+
+        return 0;
+}
+
+
+static void do_test_consume_batch_with_seek() {
+        const int partition_cnt = 2;
+        rd_kafka_queue_t *rkq;
+        const char *topic;
+        rd_kafka_t *consumer;
+        rd_kafka_topic_t *rkt;
+        int p;
+        const int timeout_ms = 10000;
+        uint64_t testid;
+        const int consume_msg_cnt = 10;
+        const int produce_msg_cnt = 8;
+        rd_kafka_conf_t *conf;
+        consumer_t consumer_args = RD_ZERO_INIT;
+        test_msgver_t mv;
+        thrd_t thread_id;
+        rd_kafka_resp_err_t err;
+        const int32_t seek_partition = 0;
+        const int64_t seek_offset = 1;
+        const int expected_msg_cnt = produce_msg_cnt - seek_offset;
+
+        SUB_TEST();
+
+        test_conf_init(&conf, NULL, 60);
+        test_conf_set(conf, "enable.auto.commit", "false");
+        test_conf_set(conf, "auto.offset.reset", "earliest");
+
+        testid = test_id_generate();
+        test_msgver_init(&mv, testid);
+
+        /* Produce messages */
+        topic = test_mk_topic_name("0137-barrier_batch_consume", 1);
+
+        for (p = 0; p < partition_cnt; p++)
+                test_produce_msgs_easy(topic, testid, p,
+                                       produce_msg_cnt / partition_cnt);
+
+        /* Create consumers */
+        consumer = test_create_consumer(topic, NULL, rd_kafka_conf_dup(conf), NULL);
+
+        test_consumer_subscribe(consumer, topic);
+        test_consumer_wait_assignment(consumer, rd_false);
+
+        /* Create generic consume queue */
+        rkq = rd_kafka_queue_get_consumer(consumer);
+
+        consumer_args.what             = "CONSUMER";
+        consumer_args.rkq              = rkq;
+        consumer_args.timeout_ms       = timeout_ms;
+        consumer_args.consume_msg_cnt  = consume_msg_cnt;
+        consumer_args.expected_msg_cnt = expected_msg_cnt;
+        consumer_args.rk               = consumer;
+        consumer_args.testid           = testid;
+        consumer_args.mv               = &mv;
+        consumer_args.test             = test_curr;
+        if (thrd_create(&thread_id, consumer_batch_queue, &consumer_args) !=
+            thrd_success)
+                TEST_FAIL("Failed to create thread for %s", "CONSUMER");
+
+        rkt = test_create_producer_topic(consumer, topic);
+        err = rd_kafka_seek(rkt, seek_partition, seek_offset, 3000);
+
+        if(err == RD_KAFKA_RESP_ERR__TIMED_OUT) {
+            TEST_FAIL("Failed to seek partition %d for topic %s to offset %lld", seek_partition, topic, seek_offset);
+        }
+
+        thrd_join(thread_id, NULL);
+
+        test_msgver_verify("CONSUME", &mv,
+                           TEST_MSGVER_ORDER | TEST_MSGVER_DUP, 0,
+                           expected_msg_cnt);
+        test_msgver_clear(&mv);
+
+        rd_kafka_topic_destroy(rkt);
```

Comment:
using these topic_t objects is generally not recommended anymore, it is typically a thing of the past and only really useful for high-performant producing.